File: JANUARY/01-01-2026/Intersection in Y Shaped Lists.java
/*
class Node {
    int data;
    Node next;

    Node(int d) {
        data = d;
        next = null;
    }
}
*/

class Solution {
    public Node intersectPoint(Node head1, Node head2) {
        Node a = head1, b = head2;
        while (a != b) {
            a = (a != null) ? a.next : head2;
            b = (b != null) ? b.next : head1;
        }
        return a;
    }
}

File: JANUARY/01-01-2026/Intersection in Y Shaped Lists.js
/**
 * @param {Node} head1
 * @param {Node} head2
 * @returns {Node}
 */

/*
class Node{
    constructor(data){
        this.data = data;
        this.next = null;
    }
}
*/
class Solution {
    intersectPoint(head1, head2) {
        let a = head1, b = head2;
        while (a !== b) {
            a = a ? a.next : head2;
            b = b ? b.next : head1;
        }
        return a;
    }
}